### PR TITLE
Add default zeroGradParameters to AbstractModule

### PR DESCRIPTION
**Pull request by SeaOfOcean**
## What changes were proposed in this pull request?
training will go wrong if layer does not zeroGradParameters

Solution: Add default zeroGradParameters to AbstractModule,
the default behavior is to zero gradParameters
It can avoid missing override zeroGradParameters

## How was this patch tested?
unit test

### CODE REVIEW

**Comment by SeaOfOcean** [comment]
jenkins pass

**Comment by yiheng** [comment]
Can you give more details about the root cause of the issue and how do you fix it? It's not easy to understand from the code change.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala [hunk 1]
```diff
@@ -258,6 +258,10 @@ object DistriOptimizer {
             val taskSize = gradLength / _subModelNumber
             val extraTask = gradLength % _subModelNumber
 
+            (0 until _subModelNumber).diff(finishedThreads).foreach(i =>
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala]: Please give more description of the fix. I don't know why make this change without more background.

**Comment by yiheng** [comment]
I think this doesn't solve the problem, e.g. a new layer with parameters may still miss override the zeroGradient.

You should change the behavior of the zeroGradient in the AbstractModule

**Comment by yiheng** [comment]
Please add more information in the description of the PR. E.g. what's the problem, what's the root cause, how to fix this

**Comment by SeaOfOcean** [comment]
updated @yiheng please help have a review, thanks

**Comment by yiheng** [comment]
LGTM

**Pull request merged**